nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment-api/src/main/java/org/nuxeo/ecm/platform/comment/api/AnnotationService.java

```diff
@@ -44,37 +46,40 @@
      *
      * @param session the core session
      * @param annotationId the annotation document model id
+     * @throws NuxeoException if no annotation was found with the given id
      * @return the annotation, or {@code null} if the annotation with the given id was not found
      */
-    Annotation getAnnotation(CoreSession session, String annotationId);
+    Annotation getAnnotation(CoreSession session, String annotationId) throws NuxeoException;
 
     /**
      * Gets all annotations for a document.
      *
      * @param session the core session
      * @param documentId the annotated document id
      * @param xpath the annotated blob xpath in the document
+     * @throws NuxeoException if no annotation was found with the given id
```

Comment:
Why not just return an empty list? After all there are no comments. Also the "no annotation was found with the given id" part is wrong, as it's the annotations about a given document.